### ISSUE TITLE
how to create auth/google_token.json

### ISSUE THREAD

**Issue by bruce-szalwinski** [llevar/flickr_to_google_photos_migration#3]
The docs mention how to create the google_credentials.json but not how to create the auth/google_token.json, referenced in line 14 of migration_util.py.  Can you explain how to do that?

**Comment by yakneens** [comment]
IIRC that file should be automatically created the first time that you go through the authorization flow, i.e. program tries to init the store using a non-existent google_token, that will force it into the authorization flow, which will populate the token file for the next time. Are you getting an error at that line?

**Comment by bruce-szalwinski** [comment]
I put two of the *.msg files in the celery/out directory and then launched using command below.  Google api complains about missing google_token.json and then celery complains about unrecognized arguments. 

     celery -A celery_migration_app worker --loglevel=debug --concurrency=1 -E | tee celery.log

```
[2019-06-25 08:47:33,312: DEBUG/MainProcess] | Worker: Preparing bootsteps.
[2019-06-25 08:47:33,314: DEBUG/MainProcess] | Worker: Building graph...
[2019-06-25 08:47:33,315: DEBUG/MainProcess] | Worker: New boot order: {Timer, Hub, Pool, Autoscaler, Beat, StateDB, Consumer}
[2019-06-25 08:47:33,335: DEBUG/MainProcess] | Consumer: Preparing bootsteps.
[2019-06-25 08:47:33,335: DEBUG/MainProcess] | Consumer: Building graph...
[2019-06-25 08:47:33,351: DEBUG/MainProcess] | Consumer: New boot order: {Connection, Events, Mingle, Tasks, Control, Heart, Gossip, Agent, event loop}
[2019-06-25 08:47:33,376: DEBUG/MainProcess] | Worker: Starting Pool
[2019-06-25 08:47:33,383: DEBUG/MainProcess] ^-- substep ok
[2019-06-25 08:47:33,384: DEBUG/MainProcess] | Worker: Starting Consumer
[2019-06-25 08:47:33,384: DEBUG/MainProcess] | Consumer: Starting Connection
[2019-06-25 08:47:33,391: INFO/MainProcess] Connected to filesystem://localhost//
[2019-06-25 08:47:33,391: DEBUG/MainProcess] ^-- substep ok
[2019-06-25 08:47:33,391: DEBUG/MainProcess] | Consumer: Starting Events
[2019-06-25 08:47:33,397: DEBUG/MainProcess] ^-- substep ok
[2019-06-25 08:47:33,397: DEBUG/MainProcess] | Consumer: Starting Tasks
[2019-06-25 08:47:33,401: DEBUG/MainProcess] ^-- substep ok
[2019-06-25 08:47:33,401: DEBUG/MainProcess] | Consumer: Starting Heart
[2019-06-25 08:47:33,402: DEBUG/MainProcess] ^-- substep ok
[2019-06-25 08:47:33,402: DEBUG/MainProcess] | Consumer: Starting event loop
[2019-06-25 08:47:33,402: DEBUG/MainProcess] Timer wake-up! Next ETA 1.0 secs.
[2019-06-25 08:47:33,403: INFO/MainProcess] celery@Bruces-MacBook-Pro-2.local ready.
[2019-06-25 08:47:33,403: DEBUG/MainProcess] basic.qos: prefetch_count->4
[2019-06-25 08:47:33,404: INFO/MainProcess] Received task: celery_migration_app.migrate_photo[dd7c95e7-f18f-45a2-ac36-70b815ac8cbf]
[2019-06-25 08:47:33,405: DEBUG/MainProcess] TaskPool: Apply <function _fast_trace_task at 0x110d2b730> (args:('celery_migration_app.migrate_photo', 'dd7c95e7-f18f-45a2-ac36-70b815ac8cbf', {'lang': 'py', 'task': 'celery_migration_app.migrate_photo', 'id': 'dd7c95e7-f18f-45a2-ac36-70b815ac8cbf', 'shadow': None, 'eta': None, 'expires': None, 'group': None, 'retries': 0, 'timelimit': [None, None], 'root_id': 'dd7c95e7-f18f-45a2-ac36-70b815ac8cbf', 'parent_id': None, 'argsrepr': "('Back to school - 2006', 'https://live.staticflickr.com/93/238431540_8ee533f99a_o.jpg', 'School Daze - 2006')", 'kwargsrepr': '{}', 'origin': 'gen34627@Bruces-MacBook-Pro-2.local', 'reply_to': 'ca94639d-4eca-36d8-9aed-f8ac30536e15', 'correlation_id': 'dd7c95e7-f18f-45a2-ac36-70b815ac8cbf', 'delivery_info': {'exchange': '', 'routing_key': 'celery', 'priority': 0, 'redelivered': None}}, b'[["Back to school - 2006", "https://live.staticflickr.com/93/238431540_8ee533f99a_o.jpg", "School Daze - 2006"], {}, {"callbacks": null, "errbacks": null, "chain": null, "chord": null}]', 'application/json', 'utf-8') kwargs:{})
[2019-06-25 08:47:33,424: INFO/MainProcess] Received task: celery_migration_app.migrate_photo[dd7c95e7-f18f-45a2-ac36-70b815ac8cbf]
[2019-06-25 08:47:33,424: DEBUG/MainProcess] TaskPool: Apply <function _fast_trace_task at 0x110d2b730> (args:('celery_migration_app.migrate_photo', 'dd7c95e7-f18f-45a2-ac36-70b815ac8cbf', {'lang': 'py', 'task': 'celery_migration_app.migrate_photo', 'id': 'dd7c95e7-f18f-45a2-ac36-70b815ac8cbf', 'shadow': None, 'eta': None, 'expires': None, 'group': None, 'retries': 0, 'timelimit': [None, None], 'root_id': 'dd7c95e7-f18f-45a2-ac36-70b815ac8cbf', 'parent_id': None, 'argsrepr': "('Back to school - 2006', 'https://live.staticflickr.com/93/238431540_8ee533f99a_o.jpg', 'School Daze - 2006')", 'kwargsrepr': '{}', 'origin': 'gen34627@Bruces-MacBook-Pro-2.local', 'reply_to': 'ca94639d-4eca-36d8-9aed-f8ac30536e15', 'correlation_id': 'dd7c95e7-f18f-45a2-ac36-70b815ac8cbf', 'delivery_info': {'exchange': '', 'routing_key': 'celery', 'priority': 0, 'redelivered': None}}, b'[["Back to school - 2006", "https://live.staticflickr.com/93/238431540_8ee533f99a_o.jpg", "School Daze - 2006"], {}, {"callbacks": null, "errbacks": null, "chain": null, "chord": null}]', 'application/json', 'utf-8') kwargs:{})
[2019-06-25 08:47:33,425: DEBUG/MainProcess] Task accepted: celery_migration_app.migrate_photo[dd7c95e7-f18f-45a2-ac36-70b815ac8cbf] pid:10275
[2019-06-25 08:47:33,427: WARNING/ForkPoolWorker-1] /Users/szalwinb/.pyenv/versions/3.6.4/lib/python3.6/site-packages/oauth2client/_helpers.py:255: UserWarning: Cannot access /Users/szalwinb/git/3rdparty/flickr_to_google_photos_migration/auth/google_token.json: No such file or directory
  warnings.warn(_MISSING_FILE_MESSAGE.format(filename))
[2019-06-25 08:47:33,430: WARNING/ForkPoolWorker-1] usage: celery [--auth_host_name AUTH_HOST_NAME] [--noauth_local_webserver]
              [--auth_host_port [AUTH_HOST_PORT [AUTH_HOST_PORT ...]]]
              [--logging_level {DEBUG,INFO,WARNING,ERROR,CRITICAL}]
[2019-06-25 08:47:33,431: WARNING/ForkPoolWorker-1] celery: error: unrecognized arguments: -A celery_migration_app worker --loglevel=debug --concurrency=1 -E
[2019-06-25 08:47:34,183: ERROR/MainProcess] Process 'ForkPoolWorker-1' pid:10275 exited with 'exitcode 2'```

Running python 3.6.4 and celery 4.3.0

    python --version
    Python 3.6.4
    celery --version
    4.3.0 (rhubarb)

**Comment by bruce-szalwinski** [comment]
I got the oauth dance to work by taking the `authorize_with_google` method out of the `migration_util.py` and put it in its own file, oauth.py.  Then executed `python oauth.py`.  The browser opened, I choose my google account and authorized the app to have the scopes given below.  The sequence completed and I now have a google_token.  We'll see what happens next.

```
from __future__ import print_function
from pathlib import Path
from oauth2client import file, client, tools


def authorize_with_google():

    # Setup the Photo v1 API
    SCOPES = 'https://www.googleapis.com/auth/photoslibrary https://www.googleapis.com/auth/photoslibrary.sharing'
    token_file = (Path().parent / "auth/google_token.json").resolve().as_posix()
    store = file.Storage(token_file)
    flow = client.flow_from_clientsecrets((Path().parent / "auth/google_credentials.json").resolve().as_posix(), SCOPES)
    creds = tools.run_flow(flow, store)
    store.put(creds)

    return creds


if __name__ == '__main__':
    authorize_with_google()
```

**Comment by mocoso** [comment]
Having read this I am still not sure how to complete this. If choose to set up 'web server' credentials, then Google Oauth system redirects back to http://localhost:8080 and there is no local server running so the process won't complete. What am I missing?

**Comment by bruce-szalwinski** [comment]
I'll take a peek at this evening and see if I can re-remember what i did here.

**Comment by mocoso** [comment]
Thanks @bruce-szalwinski, do let me know if you are able to remember

**Comment by bruce-szalwinski** [comment]
Repeating my steps resulted in success :)

```bash
python oauth.py
```

```bash
Your browser has been opened to visit:
 https://accounts.google.com/o/oauth2/auth?client_id=359724388350-di9nuude6v5mgcvfv5bck05k9aovro3v.apps.googleusercontent.com&...
```

That brought up Sign in with Google and choose an account:

<img width="375" alt="Screen Shot 2020-03-15 at 2 40 26 PM" src="https://user-images.githubusercontent.com/2782146/76709444-d1f3d180-66cc-11ea-99bc-309c648c7716.png">

I picked my google account, which brought up this screen:

<img width="657" alt="Screen Shot 2020-03-15 at 2 41 11 PM" src="https://user-images.githubusercontent.com/2782146/76709466-edf77300-66cc-11ea-997a-33ee0ee45b5d.png">

Clicked advanced, and authorized flicker-xref to access my Google Account.  Choose "Allow"

Then 

```bash
Authentication successful.
```

Also did it with 

```bash
python oauth.py --noauth_local_webserver
```

which told me to follow a link:

```bash
Go to the following link in your browser:

    https://accounts.google.com/o/oauth2/auth?client_id=359724388350-di9nuude6v5mgcvfv5bck05k9aovro3v.apps.googleusercontent.com...

Enter verification code: 
```

I followed the link, it gave me a code, and I entered it at the prompt.

It both cases, I end up with at `auth/google_token.json`.
